TEMPLATES.md

```diff
@@ -0,0 +1,103 @@
+# Using `Template`s as Execution Targets With CI Operator
```

Comment:
Nit: Top level directory docs tend to be the boilerplate stuff: README.md, LICENSE, etc. More detailed docs like this and the architecture docs should go in a Documentation directory. Then the filenames don't have to be so shouty :)